lkmpg.tex

```diff
@@ -1857,6 +1857,81 @@ \subsection{Symmetric key encryption}
 
 \samplec{examples/cryptosk.c}
 
+\section{Virtual Input Device Driver}
+\label{sec:vinput}
+vinput is a API to allow easy development of virtual input drivers.
```

Comment:
The motivation of `vinput` is to complement the documentation [Linux Input Subsystem kernel API](https://www.kernel.org/doc/html/latest/input/input_kapi.html), which lacks of complete examples. You can describe some essential concepts from the kernel docs.